mithril-aggregator/tests/create_certificate_slave.rs

```diff
@@ -0,0 +1,395 @@
+        cardano_transactions_signing_config: CardanoTransactionsSigningConfig {
+            security_parameter: BlockNumber(0),
+            step: BlockNumber(30),
+        },
+        ..Configuration::new_sample()
+    };
+    let mut master_tester =
+        RuntimeTester::build(start_time_point.clone(), master_configuration.clone()).await;
+    let master_aggregator_http_server = master_tester.expose_epoch_settings().await.unwrap();
+
+    let slave_configuration = Configuration {
+        data_stores_directory: get_test_dir("create_certificate_slave"),
+        snapshot_directory: TempDir::create("aggregator-integration", "create_certificate_slave"),
+        master_aggregator_endpoint: Some(master_aggregator_http_server.url()),
+        ..master_configuration
+    };
+    let mut slave_tester = RuntimeTester::build(start_time_point, slave_configuration).await;
+
+    comment!("Epoch: 1");
+
+    comment!("Master: create signers & declare stake distribution");
+    master_tester
+        .init_state_from_fixture(&fixture)
+        .await
+        .unwrap();
+
+    comment!("Slave: create signers & declare stake distribution");
+    slave_tester
+        .chain_observer
+        .set_signers(fixture.signers_with_stake())
+        .await;
+
+    comment!("Master: bootstrap the genesis certificate");
+    master_tester
+        .register_genesis_certificate(&fixture)
+        .await
+        .unwrap();
+
+    assert_last_certificate_eq!(
+        master_tester,
+        ExpectedCertificate::new_genesis(Epoch(1), fixture.compute_and_encode_avk())
+    );
+
+    comment!("Master: start the runtime state machine");
+    cycle!(master_tester, "ready");
+    cycle!(master_tester, "signing");
+
+    comment!("Slave: start the runtime state machine");
+    cycle_err!(slave_tester, "idle");
+    cycle_err!(slave_tester, "idle");
+
+    comment!("Master: register signers");
+    master_tester
+        .register_signers(&fixture.signers_fixture())
+        .await
+        .unwrap();
+    cycle_err!(master_tester, "signing");
+
+    comment!("Master: signers send their single signature");
+    master_tester
+        .send_single_signatures(
+            SignedEntityTypeDiscriminants::MithrilStakeDistribution,
+            &fixture.signers_fixture(),
+        )
+        .await
+        .unwrap();
+
+    comment!("Master: state machine should issue a certificate for the MithrilStakeDistribution");
+    cycle!(master_tester, "ready");
+    assert_last_certificate_eq!(
+        master_tester,
+        ExpectedCertificate::new(
+            Epoch(1),
+            StakeDistributionParty::from_signers(fixture.signers_with_stake()).as_slice(),
+            fixture.compute_and_encode_avk(),
+            SignedEntityType::MithrilStakeDistribution(Epoch(1)),
+            ExpectedCertificate::genesis_identifier(Epoch(1)),
+        )
+    );
+
+    comment!("Epoch: 2");
+
+    comment!("Master: change the epoch");
```

Comment:
Most of the file comments repeat what the code do but doesn't say why we are doing those steps, this make this already long test even more difficult to understand :sweat_smile:.
Detailing for each epochs what we want to test would already be a huge help.

Reply:
You're right, I added explanations about what we want to test at each epoch for both the master and slave aggregators :+1: